src/app/user-profile-search/checkboxes-group/checkboxes-group.component.html

```diff
@@ -0,0 +1,6 @@
+<div *ngFor="let value of values | slice : 0: countLimit">
+  <mat-checkbox (change)="onChange()" [(ngModel)]="value.checked">{{value.title}}</mat-checkbox>
```

Comment:
The `values` property is an input parameter of the component. Therefore, binding `value.checked` to the checkbox using `[(ngModel)]` will alter the state of object which was passed in from the outer component. This is considered an anti-pattern because on the outer level it is not obvious that the object will be mutated and you need an event handler anyway to respond to changes. Therefore, I'd recommend to bind a local state object here which is initialized with the values passed in as `values`. In the `onChange()` handler you should pass an object to the event emitter which contains the current selection.

Reply:
I updated the implementation, so now FilterGroupComponent emits all selected filter values on change event